yang/frr-bgp-common.yang

```diff
@@ -0,0 +1,1092 @@
+           route suppression takes place. A route is not installed
+           in the forwarding information base (FIB), or announced
+           even if it is reachable during the period that it is
+           suppressed.";
+      }
+
+      leaf unreach-decay {
+        when "../enable = 'true'";
+        type uint8 {
+          range "1..255";
+        }
+        units "seconds";
+        default "60";
+        description
+          "This value acts the same as reach-decay except that it
+           specifies the rate at which the instability metric is
+           decayed when a route is unreachable. It should have a
+           value greater than or equal to reach-decay.";
+      }
+    }
+  }
+
+  grouping flow-spec-config {
+    container flow-spec-config {
+      description
+        "Flow spec feature.";
+      leaf interface {
+        type frr-interface:interface-ref {
+          require-instance false;
+        }
+        description
+          "The local interface.";
+      }
+    }
+  }
+
+  grouping global-graceful-shutdown {
+    description
+      "Structural grouping used to include graceful-shutdown
+       configuration for both BGP neighbors and peer groups.";
+    container graceful-shutdown {
+      description
+        "BGP Graceful shutdown feature.";
+      leaf enable {
+        type boolean;
+        default "false";
+        description
+          "Enable graceful-shutdown feature.";
+      }
+    }
+  }
+
+  grouping global-filter-config {
+    description
+      "Structural grouping used to include filter
+       configuration for BGP RIB table.";
+    container filter-config {
+      description
+        "BGP table to RIB route download filter.";
+      uses rmap-policy-export;
+    }
+  }
+
+  grouping route-distinguisher-params {
+    description
+      "Route distinguisher value as per RFC4364.";
+    leaf rd {
+      type rt-types:route-distinguisher;
+      description
+        "Route distinguisher value as per RFC4364.";
+    }
+  }
+
+  grouping vpn-label-params {
+    description
+      "Label value for VRF.";
+    choice label-allocation-mode {
+      case manual {
+        leaf label-index {
+          type rt-types:mpls-label;
+          description
+            "Label index to associate with the prefix.";
+        }
+      }
+
+      case auto {
+        leaf label-auto {
+          type boolean;
+          default "false";
+          description
+            "Automatically assign a label.";
+        }
+      }
+    }
+  }
+
+  grouping vpn-nexthop-params {
+    description
+      "Specify next hop to use for VRF advertised prefixes.";
+    leaf nexthop {
+      type inet:ip-address;
+      description
+        "Nexthop IP address.";
+    }
+  }
+
+  grouping vpn-route-target-params {
+    description
+      "Route Target value.";
+    leaf redirect-rt {
+      type rt-types:route-target;
+      description
+        "Flow-spec redirect type route target.";
+    }
+
+    choice rt-direction {
```

Comment:
vrf route leak (vpn) configures list of route-targets and applies with one of the direction (import| export | both) .

Reply:
changed it to leaf-list.